prober/prober.go

```diff
@@ -12,17 +12,37 @@ import (
 	"fmt"
 	"hash/fnv"
 	"log"
+	"maps"
 	"math/rand"
 	"sync"
 	"time"
 
 	"github.com/prometheus/client_golang/prometheus"
 )
 
-// ProbeFunc is a function that probes something and reports whether
-// the probe succeeded. The provided context's deadline must be obeyed
-// for correct probe scheduling.
-type ProbeFunc func(context.Context) error
+// ProbeClass defines a probe of a specific type: a probing function that will
```

Comment:
Since you invited bikeshedding: This could be simply "Probe" since it is all the stuff one needs to conduct a probe. Or if that's too conflicty with other things, perhaps ProbeConfig. (I agree with your diagnosis that "class" doesn't quite sit right, but it's probably fine).

Reply:
I thought about this a bit. We already have a [Probe](https://github.com/tailscale/tailscale/blob/b0fbd85592c87c9d1a19b4961bd55355b1f28bd8/prober/prober.go#L173) which reflects a running probe, which is why I originally reached for a different (more specific) name. I think ProbeConfig would work as an internal struct name, but I also need to expose the type of a probe as a separate metric label, for which `class` I think is a clearer name than `config`. So I am keeping Class, sorry :)